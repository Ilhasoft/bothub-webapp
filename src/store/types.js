export default {
  SET_TOKEN: 'SET_TOKEN',
  OPEN_LOGIN_MODAL: 'OPEN_LOGIN_MODAL',
  CLOSE_LOGIN_MODAL: 'CLOSE_LOGIN_MODAL',
  SET_PROFILE: 'SET_PROFILE',
  SET_NICKNAME_AUTHENTICATED: 'SET_NICKNAME_AUTHENTICATED',
  SET_REPOSITORY_RELATED_UUID: 'SET_REPOSITORY_RELATED_UUID',
  CACHE_FETCH: 'CACHE_FETCH',
};
