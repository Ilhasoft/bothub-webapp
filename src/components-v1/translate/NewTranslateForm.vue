<template>
  <div>
    <form
      class="columns is-variable is-2"
      @submit.prevent="onSubmit()">
      <div class="column">
        <bh-field
          :errors="errors.text"
          label>
          <example-text-with-highlighted-entities-input
            ref="textInput"
            v-model="text"
            :entities="entities"
            :available-entities="repository.entities"
            placeholder="Translate sentence"
            size="medium"
            @textSelected="setTextSelected($event)" />
        </bh-field>
      </div>
      <div class="column is-narrow">
        <bh-field label>
          <bh-button
            :disabled="!isValid || submitting"
            secondary
            type="submit">Submit Translation</bh-button>
        </bh-field>
      </div>
    </form>
    <div class="columns is-variable is-1">
      <div class="column">
        <bh-field :errors="errors.entities">
          <entities-input
            ref="entitiesInput"
            v-model="entities"
            :repository="repository"
            :text="text"
            :text-selected="textSelected"
            :available-entities="repository.entities"
            custom-label-disabled
            @entityAdded="onEntityAdded($event)"
            @entityEdited="onEditEntity($event)" />
        </bh-field>
      </div>
    </div>
  </div>
</template>

<script>
import { mapActions } from 'vuex';
import ExampleTextWithHighlightedEntitiesInput from '@/components/inputs/ExampleTextWithHighlightedEntitiesInput';
import EntitiesInput from '@/components/inputs/EntitiesInput';


const components = {
  ExampleTextWithHighlightedEntitiesInput,
  EntitiesInput,
};

export default {
  name: 'NewTranslateForm',
  components,
  props: {
    extraEntitiesList: {
      type: Array,
      default: () => ([]),
    },
    exampleId: {
      type: Number,
      required: true,
    },
    translateTo: {
      type: String,
      required: true,
    },
    repository: {
      type: Object,
      default: null,
    },
  },
  data() {
    return {
      text: '',
      entities: [],
      textSelected: null,
      errors: {},
      submitting: false,
    };
  },
  computed: {
    isValid() {
      return !!this.text;
    },
  },
  watch: {
    text() {
      this.errors = {};
    },
    entities: {
      handler() {
        this.errors = {};
      },
      deep: true,
    },
  },
  methods: {
    ...mapActions([
      'newTranslation',
    ]),
    setTextSelected(value) {
      this.textSelected = value;
    },
    onEntityAdded() {
      if (this.$refs.textInput.clearSelected) {
        this.$refs.textInput.clearSelected();
      }
    },
    onEditEntity(entity) {
      if (this.$refs.textInput.emitTextSelected) {
        this.$refs.textInput.emitTextSelected({
          selectionStart: entity.start,
          selectionEnd: entity.end,
        });
      }
    },
    async onSubmit() {
      this.errors = {};
      this.submitting = true;

      try {
        await this.newTranslation({
          exampleId: this.exampleId,
          language: this.translateTo,
          text: this.text,
          entities: this.entities,
        });

        this.text = '';
        this.entities = [];

        this.submitting = false;
        this.$emit('translated');
        return true;
      } catch (error) {
        const { language, ...data } = error.response && error.response.data;

        if (data) {
          this.errors = data;
        }

        this.submitting = false;
        if (language) {
          this.$toast.open({
            message: language.join(' '),
            type: 'is-danger',
          });
        }
      }

      return false;
    },
  },
};
</script>
