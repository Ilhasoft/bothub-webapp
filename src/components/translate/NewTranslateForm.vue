<template>
  <div>
    <form
      class="columns is-variable is-2"
      @submit.prevent="onSubmit()">
      <div class="column">
        <b-field
          :errors="errors.text"
          label>
          <example-text-with-highlighted-entities-input
            id="tour-translate-step-4"
            ref="textInput"
            :is-previous-disabled="true"
            v-model="text"
            :is-step-blocked="text.length === 0"
            :entities="entities"
            :available-entities="repository.entities"
            :placeholder="$t('webapp.translate.translate_sentence')"
            size="normal"
            @textSelected="setTextSelected($event)" />
        </b-field>
      </div>
      <div class="column is-narrow">
        <b-field>
          <b-button
            id="tour-translate-step-5"
            :is-next-disabled="true"
            :is-previous-disabled="true"
            :disabled="!isValid || submitting"
            type="is-secondary"
            native-type="submit">{{ $t('webapp.translate.submit_translation') }}</b-button>
        </b-field>
      </div>
    </form>
    <div class="columns is-variable is-1">
      <div class="column">
        <b-field :message="errors.entities">
          <entities-input
            ref="entitiesInput"
            v-model="entities"
            :repository="repository"
            :text="text"
            :text-selected="textSelected"
            :available-entities="repository.entities"
            :available-add-label="false"
            @entityAdded="onEntityAdded($event)"
            @entityEdited="onEditEntity($event)" />
        </b-field>
      </div>
    </div>
  </div>
</template>

<script>
import { mapActions } from 'vuex';
import ExampleTextWithHighlightedEntitiesInput from '@/components/inputs/ExampleTextWithHighlightedEntitiesInput';
import EntitiesInput from '@/components/inputs/EntitiesInput';


const components = {
  ExampleTextWithHighlightedEntitiesInput,
  EntitiesInput,
};

export default {
  name: 'NewTranslateForm',
  components,
  props: {
    extraEntitiesList: {
      type: Array,
      default: () => ([]),
    },
    exampleId: {
      type: Number,
      required: true,
    },
    translateTo: {
      type: String,
      required: true,
    },
    repository: {
      type: Object,
      default: null,
    },
  },
  data() {
    return {
      text: '',
      entities: [],
      textSelected: null,
      errors: {},
      submitting: false,
      blockedNextStepTutorial: true,
    };
  },
  computed: {
    isValid() {
      return !!this.text;
    },
  },
  watch: {
    text() {
      this.errors = {};
    },
    entities: {
      handler() {
        this.errors = {};
      },
      deep: true,
    },
  },
  methods: {
    ...mapActions([
      'newTranslation',
    ]),
    setTextSelected(value) {
      this.textSelected = value;
    },
    onEntityAdded() {
      if (this.$refs.textInput.clearSelected) {
        this.$refs.textInput.clearSelected();
      }
    },
    onEditEntity(entity) {
      if (this.$refs.textInput.emitTextSelected) {
        this.$refs.textInput.emitTextSelected({
          selectionStart: entity.start,
          selectionEnd: entity.end,
        });
      }
    },
    async onSubmit() {
      this.errors = {};
      this.submitting = true;
      try {
        await this.newTranslation({
          exampleId: this.exampleId,
          language: this.translateTo,
          text: this.text,
          entities: this.entities,
        });

        this.text = '';
        this.entities = [];

        this.submitting = false;
        this.$emit('translated');
        return true;
      } catch (error) {
        const { language, ...data } = error.response && error.response.data;

        if (data) {
          this.errors = data;
        }

        this.submitting = false;
        if (language) {
          this.$buefy.notification.open({
            message: language.join(' '),
            type: 'is-danger',
          });
        }
      }

      this.blockedNextStepTutorial = !this.blockedNextStepTutorial;
      this.$emit('eventStep');

      return false;
    },
  },
};
</script>
