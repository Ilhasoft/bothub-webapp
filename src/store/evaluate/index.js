import actions from './actions';
import state from './state';
import mutations from './mutations';
import getters from './getters';
import types from './types';

export default {
  actions,
  state,
  mutations,
  getters,
  types,
};
