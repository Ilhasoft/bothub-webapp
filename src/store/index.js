import Vue from 'vue';
import Vuex from 'vuex';

import Auth from './auth';
import User from './user';
import Org from './org';
import Repository from './repository';
import Category from './category';
import Example from './example';
import EvaluateExample from './evaluate-example';
import Translate from './translate';
import CachedFetch from './cached-fetch';
import Entity from './entity';
import Intent from './intent';
import News from './news';
import Tutorial from './tutorial';
import External from './external';

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    Auth,
    User,
    Org,
    Repository,
    Category,
    Example,
    Translate,
    CachedFetch,
    EvaluateExample,
    Entity,
    News,
    Tutorial,
    External,
    Intent,
  },
});

store.dispatch('retriveAuthToken');

export default store;
