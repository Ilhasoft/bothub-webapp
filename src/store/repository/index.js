import state from './state';
import getters from './getters';
import actions from './actions';
import mutations from './mutations';
import types from './types';

export default {
  state,
  getters,
  actions,
  mutations,
  types,
};
