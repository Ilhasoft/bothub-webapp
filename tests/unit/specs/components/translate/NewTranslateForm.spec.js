/* eslint-disable import/first */
jest.mock('@/api/request');


import { shallowMount, createLocalVue } from '@vue/test-utils';

import NewTranslateForm from '@/components/translate/NewTranslateForm';
import store from '@/store';


const localVue = createLocalVue();


describe('NewTranslateForm.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(NewTranslateForm, {
      localVue,
      mocks: {
        $t: () => 'some specific text',
      },
      propsData: {
        exampleId: 1,
        translateTo: 'pt',
        repository: {},
      },
      store,
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('isValid is false', () => {
    expect(wrapper.vm.isValid).toBeFalsy();
  });

  describe('fill with valid data', () => {
    beforeEach(() => {
      wrapper.vm.text = 'meu nome é douglas';
      wrapper.vm.entities = [
        {
          start: 12,
          end: 19,
          entity: 'name',
        },
      ];
    });

    test('isValid is true', () => {
      expect(wrapper.vm.isValid).toBeTruthy();
    });

    // describe('on submit', () => {
    //   let r;
    //   beforeEach(async () => {
    //     r = await wrapper.vm.onSubmit();
    //   });

    //   test('return is true', () => {
    //     expect(r).toBeTruthy();
    //   });

    //   test('created event emitted', () => {
    //     expect(wrapper.emitted('translated')).toBeDefined();
    //   });
    // });
  });

  describe('fill with invalid data', () => {
    beforeEach(() => {
      wrapper.vm.text = 'meu nome é douglas';
      wrapper.vm.entities = [];
    });

    // describe('on submit', () => {
    //   let r;
    //   beforeEach(async () => {
    //     r = await wrapper.vm.onSubmit();
    //   });

    //   test('return is false', () => {
    //     expect(r).toBeFalsy();
    //   });
    // });
  });

  describe('entitiesInput emit entityAdded', () => {
    let entitiesInput;
    beforeEach(() => {
      entitiesInput = wrapper.find({ ref: 'entitiesInput' });
      entitiesInput.vm.$emit('entityAdded', {});
    });

    test('entityAdded event emitted by entitiesInput', () => {
      expect(entitiesInput.emitted('entityAdded')).toBeDefined();
    });
  });

  describe('entitiesInput emit entityEdited', () => {
    let entitiesInput;
    beforeEach(() => {
      entitiesInput = wrapper.find({ ref: 'entitiesInput' });
      entitiesInput.vm.$emit('entityEdited', {});
    });

    test('entityEdited event emitted by entitiesInput', () => {
      expect(entitiesInput.emitted('entityEdited')).toBeDefined();
    });
  });

  describe('textInput emit textSelected', () => {
    let textInput;
    const eventValue = { start: 0, end: 3 };

    beforeEach(() => {
      textInput = wrapper.find({ ref: 'textInput' });
      textInput.vm.$emit('textSelected', eventValue);
    });

    test('textSelected event emitted by entitiesInput', () => {
      expect(textInput.emitted('textSelected')).toBeDefined();
    });

    test('textSelected is equal to value emitted by textInput', () => {
      expect(wrapper.vm.textSelected).toMatchObject(eventValue);
    });
  });
});
