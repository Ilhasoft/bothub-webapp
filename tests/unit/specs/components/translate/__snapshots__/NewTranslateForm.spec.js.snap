// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTranslateForm.vue renders correctly 1`] = `
<div>
  <form class="columns is-variable is-2">
    <div class="column">
      <b-field label="">
        <example-text-with-highlighted-entities-input-stub value="" entities="" id="tour-translate-step-4" is-previous-disabled="true" is-step-blocked="true" placeholder="some specific text" size="normal"></example-text-with-highlighted-entities-input-stub>
      </b-field>
    </div>
    <div class="column is-narrow">
      <b-field>
        <b-button id="tour-translate-step-5" is-next-disabled="true" is-previous-disabled="true" disabled="true" type="is-secondary" native-type="submit">some specific text</b-button>
      </b-field>
    </div>
  </form>
  <div class="columns is-variable is-1">
    <div class="column">
      <b-field>
        <entities-input-stub value="" repository="[object Object]" text="" availableentities="" entitiesforedit=""></entities-input-stub>
      </b-field>
    </div>
  </div>
</div>
`;
